docroot/profiles/custom/cgov_site/modules/custom/cgov_core/cgov_core.module

```diff
@@ -601,4 +601,28 @@ function cgov_core_preprocess_image_formatter(&$variables) {
   ) {
     $variables['image']['#alt'] = "";
   }
+
+}
+
+/**
+ * Implements hook_entity_build_defaults_alter().
+ */
+function cgov_core_entity_build_defaults_alter(array &$build, EntityInterface $entity, $view_mode) {
+
+  // Add the bundle as a cache tag to help simpflify cache invalidation.
```

Comment:
When an item is rendering as the full view-mode it will have both the :full and :all-view-modes cache tag.

Otherwise the rendered item will just contain :all-view-modes tag.

Reply:
I think this will work. I would request that the actual build array be modified only in one place...
```
$tags = $view_mode == 'full' ? [$entity->bundle() . ':full'] : [];
$tags[] = $entity->bundle() . ':all-view-modes';
$build['#cache'] ...
```